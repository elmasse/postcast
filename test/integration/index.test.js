/* global describe, test, expect */
import React from 'react'
import fetchMock from 'fetch-mock'

import { mount } from 'enzyme'

/* SUT */
import Postcast from '../../src'

describe('Postcast', () => {
  test('load src', async () => {
    fetchMock.get('/source', {
      body: '# This is a title\n And this a paragraph',
      status: 200,
      headers: {
        'content-type': 'text/markdown'
      }
    })

    const wrapper = mount(<Postcast src={'/source'} />)

    await wrapper.instance().componentDidMount()

    wrapper.update()

    expect(wrapper.state('loaded')).toBe(true)
    expect(wrapper.state('error')).toBeFalsy()

    expect(wrapper).toMatchSnapshot()
  })

  test('load src with error', async () => {
    fetchMock.get('/not-found', {
      status: 404
    })

    const wrapper = mount(<Postcast src={'/not-found'} />)

    await wrapper.instance().componentDidMount()

    wrapper.update()

    expect(wrapper.state('loaded')).toBe(false)
    expect(wrapper.state('error')).not.toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  test('load src with non markdown endpoint', async () => {
    fetchMock.get('/not-markdown', {
      body: '<html></html>',
      status: 200,
      headers: {
        'content-type': 'text/html'
      }
    })

    const wrapper = mount(<Postcast src={'/not-markdown'} />)

    await wrapper.instance().componentDidMount()

    wrapper.update()

    expect(wrapper.state('loaded')).toBe(false)
    expect(wrapper.state('error')).not.toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  test('markdown as children', async () => {
    const wrapper = mount(<Postcast >{() => `# Title\nAnd Paragraph`}</Postcast>)

    await wrapper.instance().componentDidMount()

    wrapper.update()

    expect(wrapper.state('loaded')).toBe(true)
    expect(wrapper.state('error')).toBeFalsy()

    expect(wrapper).toMatchSnapshot()
  })
})
