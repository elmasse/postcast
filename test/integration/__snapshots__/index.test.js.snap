// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Postcast load src 1`] = `
<Postcast
  src="/source"
>
  <Styled(div)>
    <div
      className="css-2iy19l"
    >
      <Player
        markdown="# This is a title
 And this a paragraph"
      >
        <Styled(div)
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="css-o636u2"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <Runner
              frame={
                <Frame>
                  <Content>
                    <h1>
                      This is a title
                    </h1>
                  </Content>
                </Frame>
              }
              metadata={null}
              onEnd={[Function]}
              pause={true}
              play={false}
            >
              <NoOp
                duration={2000}
                key="runner-h-2"
                lang=""
                onEnd={[Function]}
                pause={true}
                play={false}
                text="This is a title"
              />
            </Runner>
            <Frame
              captions={true}
            >
              <Styled(div)>
                <div
                  className="css-8z9imq"
                >
                  <Content
                    key="h-2"
                  >
                    <Styled(div)
                      css="css-5j5vrg"
                    >
                      <div
                        className="css-1uahr9q"
                      >
                        <h1
                          key="h-3"
                        >
                          This is a title
                        </h1>
                      </div>
                    </Styled(div)>
                  </Content>
                </div>
              </Styled(div)>
            </Frame>
            <Styled(div)
              hide={false}
            >
              <div
                className="css-15dbzkp"
              >
                <Styled(div)>
                  <div
                    className="css-lmwhk0"
                  >
                    <Play
                      onPause={[Function]}
                      onPlay={[Function]}
                      playing={false}
                      size={30}
                    >
                      <svg
                        height={30}
                        onClick={[Function]}
                        onPause={[Function]}
                        onPlay={[Function]}
                        viewBox="0 0 48 48"
                        width={30}
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <title>
                          Play
                        </title>
                        <g>
                          <path
                            d="M20 33l12-9-12-9v18zm4-29C12.95 4 4 12.95 4 24s8.95 20 20 20 20-8.95 20-20S35.05 4 24 4zm0 36c-8.82 0-16-7.18-16-16S15.18 8 24 8s16 7.18 16 16-7.18 16-16 16z"
                          />
                        </g>
                      </svg>
                    </Play>
                    <Captions
                      captions={true}
                      onToggle={[Function]}
                      size={30}
                    >
                      <svg
                        fill="#000"
                        height={30}
                        onClick={[Function]}
                        viewBox="0 0 48 48"
                        width={30}
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <title>
                          Set Captions 
                          off
                        </title>
                        <path
                          d="M38 8H10c-2.21 0-4 1.79-4 4v24c0 2.21 1.79 4 4 4h28c2.21 0 4-1.79 4-4V12c0-2.21-1.79-4-4-4zM22 22h-3v-1h-4v6h4v-1h3v2c0 1.1-.89 2-2 2h-6c-1.11 0-2-.9-2-2v-8c0-1.1.89-2 2-2h6c1.11 0 2 .9 2 2v2zm14 0h-3v-1h-4v6h4v-1h3v2c0 1.1-.89 2-2 2h-6c-1.11 0-2-.9-2-2v-8c0-1.1.89-2 2-2h6c1.11 0 2 .9 2 2v2z"
                        />
                      </svg>
                    </Captions>
                    <Language
                      size={30}
                    >
                      <Styled(div)
                        fill="black"
                        size={30}
                      >
                        <div
                          className="css-up86ey"
                          fill="black"
                          size={30}
                        >
                          <svg
                            fill="black"
                            height={21}
                            viewBox="0 0 24 24"
                            width={21}
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
                            />
                          </svg>
                          en-US
                        </div>
                      </Styled(div)>
                    </Language>
                  </div>
                </Styled(div)>
                <Timeline
                  active={0}
                  frames={
                    Array [
                      <Frame>
                        <Content>
                          <h1>
                            This is a title
                          </h1>
                        </Content>
                      </Frame>,
                      <Frame>
                        <Content>
                          <h1>
                            This is a title
                          </h1>
                        </Content>
                        <Caption>
                          <p>
                             And this a paragraph
                          </p>
                        </Caption>
                      </Frame>,
                    ]
                  }
                  onChangeFrame={[Function]}
                >
                  <Styled(input)
                    max={1}
                    onChange={[Function]}
                    type="range"
                    value={0}
                  >
                    <input
                      className="css-dig042"
                      max={1}
                      onChange={[Function]}
                      type="range"
                      value={0}
                    />
                  </Styled(input)>
                </Timeline>
              </div>
            </Styled(div)>
          </div>
        </Styled(div)>
      </Player>
    </div>
  </Styled(div)>
</Postcast>
`;

exports[`Postcast load src with error 1`] = `
<Postcast
  src="/not-found"
>
  <Styled(div)>
    <div
      className="css-2iy19l"
    >
      <Styled(div)>
        <div
          className="css-72j8j7"
        >
          <h2>
            Something went wrong
          </h2>
          Looks like we cannot connect to /not-found
        </div>
      </Styled(div)>
    </div>
  </Styled(div)>
</Postcast>
`;

exports[`Postcast load src with non markdown endpoint 1`] = `
<Postcast
  src="/not-markdown"
>
  <Styled(div)>
    <div
      className="css-2iy19l"
    >
      <Styled(div)>
        <div
          className="css-72j8j7"
        >
          <h2>
            Something went wrong
          </h2>
          /not-markdown does not look like a markdown file!
        </div>
      </Styled(div)>
    </div>
  </Styled(div)>
</Postcast>
`;

exports[`Postcast markdown as children 1`] = `
<Postcast>
  <Styled(div)>
    <div
      className="css-2iy19l"
    >
      <Player
        markdown="# Title
And Paragraph"
      >
        <Styled(div)
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="css-o636u2"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <Runner
              frame={
                <Frame>
                  <Content>
                    <h1>
                      Title
                    </h1>
                  </Content>
                </Frame>
              }
              metadata={null}
              onEnd={[Function]}
              pause={true}
              play={false}
            >
              <NoOp
                duration={2000}
                key="runner-h-2"
                lang=""
                onEnd={[Function]}
                pause={true}
                play={false}
                text="Title"
              />
            </Runner>
            <Frame
              captions={true}
            >
              <Styled(div)>
                <div
                  className="css-8z9imq"
                >
                  <Content
                    key="h-2"
                  >
                    <Styled(div)
                      css="css-5j5vrg"
                    >
                      <div
                        className="css-1uahr9q"
                      >
                        <h1
                          key="h-3"
                        >
                          Title
                        </h1>
                      </div>
                    </Styled(div)>
                  </Content>
                </div>
              </Styled(div)>
            </Frame>
            <Styled(div)
              hide={false}
            >
              <div
                className="css-15dbzkp"
              >
                <Styled(div)>
                  <div
                    className="css-lmwhk0"
                  >
                    <Play
                      onPause={[Function]}
                      onPlay={[Function]}
                      playing={false}
                      size={30}
                    >
                      <svg
                        height={30}
                        onClick={[Function]}
                        onPause={[Function]}
                        onPlay={[Function]}
                        viewBox="0 0 48 48"
                        width={30}
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <title>
                          Play
                        </title>
                        <g>
                          <path
                            d="M20 33l12-9-12-9v18zm4-29C12.95 4 4 12.95 4 24s8.95 20 20 20 20-8.95 20-20S35.05 4 24 4zm0 36c-8.82 0-16-7.18-16-16S15.18 8 24 8s16 7.18 16 16-7.18 16-16 16z"
                          />
                        </g>
                      </svg>
                    </Play>
                    <Captions
                      captions={true}
                      onToggle={[Function]}
                      size={30}
                    >
                      <svg
                        fill="#000"
                        height={30}
                        onClick={[Function]}
                        viewBox="0 0 48 48"
                        width={30}
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <title>
                          Set Captions 
                          off
                        </title>
                        <path
                          d="M38 8H10c-2.21 0-4 1.79-4 4v24c0 2.21 1.79 4 4 4h28c2.21 0 4-1.79 4-4V12c0-2.21-1.79-4-4-4zM22 22h-3v-1h-4v6h4v-1h3v2c0 1.1-.89 2-2 2h-6c-1.11 0-2-.9-2-2v-8c0-1.1.89-2 2-2h6c1.11 0 2 .9 2 2v2zm14 0h-3v-1h-4v6h4v-1h3v2c0 1.1-.89 2-2 2h-6c-1.11 0-2-.9-2-2v-8c0-1.1.89-2 2-2h6c1.11 0 2 .9 2 2v2z"
                        />
                      </svg>
                    </Captions>
                    <Language
                      size={30}
                    >
                      <Styled(div)
                        fill="black"
                        size={30}
                      >
                        <div
                          className="css-up86ey"
                          fill="black"
                          size={30}
                        >
                          <svg
                            fill="black"
                            height={21}
                            viewBox="0 0 24 24"
                            width={21}
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
                            />
                          </svg>
                          en-US
                        </div>
                      </Styled(div)>
                    </Language>
                  </div>
                </Styled(div)>
                <Timeline
                  active={0}
                  frames={
                    Array [
                      <Frame>
                        <Content>
                          <h1>
                            Title
                          </h1>
                        </Content>
                      </Frame>,
                      <Frame>
                        <Content>
                          <h1>
                            Title
                          </h1>
                        </Content>
                        <Caption>
                          <p>
                            And Paragraph
                          </p>
                        </Caption>
                      </Frame>,
                    ]
                  }
                  onChangeFrame={[Function]}
                >
                  <Styled(input)
                    max={1}
                    onChange={[Function]}
                    type="range"
                    value={0}
                  >
                    <input
                      className="css-dig042"
                      max={1}
                      onChange={[Function]}
                      type="range"
                      value={0}
                    />
                  </Styled(input)>
                </Timeline>
              </div>
            </Styled(div)>
          </div>
        </Styled(div)>
      </Player>
    </div>
  </Styled(div)>
</Postcast>
`;
